trl/trainer/dpo_trainer.py

```diff
@@ -1258,7 +1497,7 @@ def concatenated_forward(
             per_token_logps_[attention_mask.bool()] = per_token_logps
             per_token_logps = per_token_logps_
 
-        all_logps = per_token_logps.sum(-1)
+        all_logps = per_token_logps[:, 1:].sum(-1)
```

Comment:
is it a bug from DPO in main?

Reply:
@qgallouedec @kashif was it a bug?

I think it's due to the roll just above.